File: EquationArranger/src/main/java/Quadratic/Equation/Tests/Scanner/ScanState.java
package Quadratic.Equation.Tests.Scanner;

public enum ScanState
{
    DuringStart,
    DuringNumber,
    DuringVariable,
    DuringCoefficientSign,
    DuringExponentSign,
    DuringExponentNumber,
    DuringEqualSign,
    DuringMultiplySign,
}

File: src/equationscanner/scan_state.go
package equationscanner

type ScanState int

const (
    DuringStart ScanState =      	iota
    DuringNumber
    DuringVariable 
    DuringCoefficientSign
    DuringExponentSign
    DuringExponentNumber
    DuringEqualSign
    DuringMultiplySign
)